### PR TITLE
Bugfix/undo lemmatising

### PR DESCRIPTION
**Pull request by raquel-ucl**
Need to test one more case, but I can't because the ORACC server seems down. Please wait for that before merging.

This changes include #90 as well as fixes for undoing lemmatising.

### CODE REVIEW

**Comment by raquel-ucl** [comment]
Ready to merge now.

File: python/nammu/SOAPClient/SOAPClient.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def setup_logger(self):
 
 
     def create_request(self, **kwargs):
-        url = self.url + ":" + self.port
+        url = "{}:{}".format(self.url, self.port)
```
Comment (jenshnielsen) [inline python/nammu/SOAPClient/SOAPClient.py]: 👍

File: python/nammu/controller/AtfAreaController.py [hunk 2]
```diff
@@ -15,16 +15,24 @@ class AtfAreaController(object):
...
+
     def setAtfAreaText(self, text):
+        '''
+        Need to manually make the edit compound start because the setText()
+        method triggers 2 consecutive INSERT events: one to set the text to "" 
+        and another to set the text to the given text.
+        Default behaviour of the compound edit would create a new edit for each
+        INSERT event, so we have to manually group those two together.
+        '''
+        self.view.edit_listener.compound()
```
Comment (jenshnielsen) [inline python/nammu/controller/AtfAreaController.py]: Would be nice if this was a context manager (with self.view.edit_listener.compound() as bla) but I guess this is's supported by Jython/Swing

File: python/nammu/view/AtfAreaView.py [hunk 3]
```diff
@@ -274,15 +274,49 @@ def keyTyped(self, ke):
...
+        if (edit_type == "INSERT" or edit_type == "REMOVE") \
+            and not self.must_compound:
+            # Explicitly end compound edits so their inProgress flag goes to
+            # false. Note undo() only undoes compound edits when they are not
+            # in progress.
+            self.current_compound.end()
+            self.current_compound = CompoundEdit()
+            self.undo_manager.addEdit(self.current_compound)
+            
+        # Always add current edit to current compound
```
Comment (jenshnielsen) [inline python/nammu/view/AtfAreaView.py]: I think a context manger like the following should work but it's completly untested



```python
import contextlib

@contextlib.contextmanager
def force_compound(self):
    self.must_compound = False
    self.current_compound.end()
    self.undo_manager.addEdit(self.current_compound)  
    try:
        yield
    finally:
        self.must_compound = False
        self.current_compound.end()
        self.undo_manager.addEdit(self.current_compound)    
```

**Comment by jenshnielsen** [comment]
Otherwise looks good and seems to work for me locally.

File: python/nammu/view/AtfAreaView.py (hunk 3 above)
Comment (raquel-ucl) [inline python/nammu/view/AtfAreaView.py]: Just copy pasted and worked as is!

**Pull request merged**